### ISSUE TITLE
Stun lock effectiveness via Ancients may be too high

### ISSUE THREAD

**Issue by nvs** [nvs/gem#253]
With the `1.3` release and the new stun system Ancients were reworked. However, it seems that being able to stun lock has become too easy. The old style of stun added a bit of variability to the wait durations due to the use of `TriggerSleepAction ()`. The use of `PauseUnit ()` on the Ancients themselves may have also introduced a bit of inaccuracy to the delays (perhaps with the time needed to finish an existing attack). The new system, however, is always accurate. And upon being rearmed, the unit begins a fresh attack immediately.

In short, it is much easier to achieve complete stun lock as of `1.3`. This should probably be addressed.

**Comment by nvs** [comment]
Yeah, stun is too accurate now. It would be best to introduce a small variability of perhaps +/- 0.10 seconds, give or take. Testing obviously will be needed. But this should reintroduce the ineffectiveness of stun lock from previous versions. Not that it was poor by any stretch of the imagination.

**Comment by nvs** [comment]
A rework to the design of Ancients in general should address this issue.

**Comment by nvs** [comment]
And we're back for Round 2. The work done in #421 initially considered having the stun/disarm time adjust according to the number of Ancients. This was misguided. As a result, it was decided to not touch the stun and disarm timings.

Well, a while ago I told someone that to fix Ancients I needed to make a single one be able to stunlock. In the previous attempt, I was too focused on trying to have the timings be adjusted by the number of Ancients previously.  But these timings should remain static. Thus, clustering Ancients increases the odds of stunlock. But three Ancients should not reach the level of stunlock that exists now. Just need to crunch some numbers.

**Comment by nvs** [comment]
Ah, a bit of information about Ancient timings. I did this testing a month or two ago, but did not include the results anywhere. It seems that the attack cooldown is roughly `0.11` seconds. Attack speed modifications (e.g. Opal auras) seem to have no effect whatsoever. The perceived effectiveness of stunlock in a game comes down to sheer luck in regards to the alignment of Ancient attack cooldowns. At worst, all will be aligned and the `0.11` cooldown will be fully felt. At best, this cooldown will not be perceived at all.

Another bit of information to realize is that increasing the value of a single Ancient, by giving it the chance to stun multiple times in succession, does nothing to lessen the value of a cluster of Ancients. If anything, it increases it further. The solution, as I see it, is to embrace the attack cooldown of Ancients, but extend it to stuns in general. For example, a new rule that says that a unit cannot be stunned within `0.1 `seconds of a stun having worn off. This rule prevents a cluster from ever approaching true stunlock.

That rule, combined with adjusted Ancient timings, should achieve the desired result: a single Ancient of sometimes getting chain stuns, and true stunlock effectively being removed from the game.
